src/libutil/config.cc

```diff
@@ -170,9 +222,22 @@ void AbstractConfig::applyConfig(const std::string & contents, const std::string
             set(name, value);
 
     // Then apply other settings
-    for (const auto & [name, value] : parsedContents)
-        if (name != "experimental-features" && name != "extra-experimental-features")
+    // XXX: NIX_PATH must override the regular setting!
+    // Environment variables overriding settings should probably be part of the Config mechanism,
+    // but at the time of writing it's not worth building that for just one thing
+    bool overrideNixPath = false;
+    if (auto nixPathEnv = getEnv("NIX_PATH")) {
+        overrideNixPath = true;
+        set("nix-path", parseNixPath(nixPathEnv.value()));
```

Comment:
If no config whatsoever is loaded, e.g. in a highly isolated environment like the Nix build sandbox, then `loadConfFile` will never call `applyConfig`, and this won't run.
I think this particular line could be moved to the `getDefaultNixPath` function, or into `loadConfFile`, duplicating the `if` condition as well.

Reply:
Ah I see my mistake now.